winpr/libwinpr/utils/asn1/asn1.c

```diff
@@ -1377,6 +1377,32 @@ size_t WinPrAsn1DecReadContextualOID(WinPrAsn1Decoder* dec, WinPrAsn1_tagId tagI
 	return ret;
 }
 
+size_t WinPrAsn1DecReadContextualOctetString(WinPrAsn1Decoder* dec, WinPrAsn1_tagId tagId,
+                                             BOOL* error, WinPrAsn1_OctetString* target,
+                                             BOOL allocate)
+{
+	size_t ret, ret2;
+	WinPrAsn1_tag ftag;
+	WinPrAsn1Decoder content;
+
+	WINPR_ASSERT(error);
+
+	ret = WinPrAsn1DecPeekContextualTag(dec, &ftag, &content);
+	if (!ret || ftag != tagId)
+		return 0;
```

Comment:
I think the bug is present in other `WinPrAsn1DecReadContextualXXXXX` functions: `*error` is not set in that case.

Reply:
Would be nice to complement asn1 unitary tests to check for that case.

Reply:
Do you want to do that in a separate PR?

Reply:
I will